### DISCUSSION TITLE
Problema de despliegue. ¿En las urls hay que poner localhost o el nombre del contenedor?

### DISCUSSION THREAD

**Discussion by samuel-bustamante** [Arquisoft/faq#60] [Q&A] [answered]
En local, nuestra aplicación despliega sin problemas, pero cuando la deplegamos en la máquina virtual los contenedores no se comunican correctamente y aparecen problemas con las rutas. Mi duda es si aquí, por ejemplo:
const userServiceUrl = process.env.USER_SERVICE_URL || 'http://localhost:8001';
sería necesario poner localhost o el nombre del contenedor. Actualmente usamos una variable de entorno en la que usamos el nombre del contenedor y si no está definida usamos localhost.
No entiendo porque no es capaz de resolver esta dirección en despliegue. Gracias un saludo.

**Comment by pglez82** [comment, marked as answer]
Buenas, si lo tenéis configurado como estaba originalmente, cuando desplegáis sin usar docker, la variable USER_SERVICE_URL no debería de estar definida y por tanto debería de apuntar a localhost. Es decir, tendríais corriendo en vuestra máquina, en el puerto 8001 el userservice. En el caso de usar docker, lo normal es acceder con el nombre del contenedor ya que los contenedores están configurados para compartir la misma red (ver el docker-compose). Es por esto que en el docker-compose se le pasa al gateway service la siguiente configuración:

```yaml
networks:
      - mynetwork
environment:
      AUTH_SERVICE_URL: http://authservice:8002
      USER_SERVICE_URL: http://userservice:8001
```

haciendo que esas variables de entorno existan y que por tanto el gatewayservice las use para acceder a los otros servicios. 

Hay otras opciones como poner que los contenedores usen directamente la red del host y ocupen un puerto del host. En ese caso, podría funcionar lo de localhost incluso usando contenedores en una arquitectura con docker-compose, y no harían falta esas variables de entorno.
